api_gen.py

```diff
@@ -238,7 +240,24 @@ def write_cython_imp(self):
             raise ValueError("Return code <<%s>> unknown" % self.line.code)
 
         # Have to use except * because Cython can't handle special types here
-        imp = """\
+        if self.line.nogil:
+            imp = """\
+cdef {0.code} {0.fname}({0.sig}) except *:
+    cdef {0.code} r
+    with nogil:
+        _hdf5.H5Eset_auto(NULL, NULL)
```

Comment:
Should this be using `set_default_error_handler`? I don't think we provide an API to set the handler to a Python function, so it should always be NULL or `H5Eprint`, and thus safe to use with nogil.

Reply:
No idea, I just reproduced what was present.

Reply:
@t20100 changed what was present in #1353, though.